code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/integration/azure/AzureTokenCreator.java

```diff
@@ -51,4 +51,24 @@ public void reset() {
     public String getName() {
         return AzureConstants.TOKEN_CREATOR;
     }
+
+    @Override
+    public Token getCurrentToken() {
+        Token token = new Token();
+
+        if (azureTokenUtils.findAndSetJWT()) {
+            Token userToken = azureTokenUtils.getUserIdentityFromJWT();
+            if(userToken != null) {
+                token.setUserIdentity(userToken.getUserIdentity());
+                token.setUserType(userToken.getUserType());
+                token.setId(userToken.getId());
+            }
+        }
+        return token;
+    }
+
+    @Override
+    public String getJWTUtil() {
+        return azureTokenUtils.getJWT();
+    }
```

Comment:
You don't have to implement entire getCurrentToken() in each TokenCreator. Each TokenCreator is having the exact same code. 
Instead add this logic to a method retrieveCurrentToken() to AbstractTokenUtil and then just invoke that method (azureTokenUtils.retrieveCurrentToken()) from each TokenCreator:: getCurrentToken()